pkg/scheduler/actions/enqueue/enqueue_test.go

```diff
@@ -0,0 +1,101 @@
+	"volcano.sh/volcano/pkg/scheduler/conf"
+	"volcano.sh/volcano/pkg/scheduler/framework"
+	"volcano.sh/volcano/pkg/scheduler/plugins/drf"
+	"volcano.sh/volcano/pkg/scheduler/plugins/gang"
+	"volcano.sh/volcano/pkg/scheduler/plugins/proportion"
+	"volcano.sh/volcano/pkg/scheduler/plugins/sla"
+	"volcano.sh/volcano/pkg/scheduler/uthelper"
+	"volcano.sh/volcano/pkg/scheduler/util"
+)
+
+func TestEnqueue(t *testing.T) {
+	plugins := map[string]framework.PluginBuilder{
+		drf.PluginName:        drf.New,
+		gang.PluginName:       gang.New,
+		sla.PluginName:        sla.New,
+		proportion.PluginName: proportion.New,
+	}
+	options.Default()
+	tests := []uthelper.TestCommonStruct{
+		{
+			Name: "when podgroup status is inqueue",
+			PodGroups: []*schedulingv1.PodGroup{
+				util.BuildPodGroup("pg1", "c1", "c1", 0, nil, schedulingv1.PodGroupInqueue),
+			},
+			Pods: []*v1.Pod{
+				util.BuildPod("c1", "p1", "", v1.PodPending, api.BuildResourceList("1", "1G"), "pg1", make(map[string]string), make(map[string]string)),
+				util.BuildPod("c1", "p2", "", v1.PodPending, api.BuildResourceList("1", "1G"), "pg1", make(map[string]string), make(map[string]string)),
+			},
+			Queues: []*schedulingv1.Queue{
+				util.BuildQueue("c1", 1, nil),
+				util.BuildQueue("c2", 1, nil),
+			},
+		},
+		{
+			Name: "when podgroup status is pending",
+			PodGroups: []*schedulingv1.PodGroup{
+				util.BuildPodGroup("pg1", "c1", "c1", 0, nil, schedulingv1.PodGroupPending),
+				util.BuildPodGroup("pg2", "c1", "c2", 0, nil, schedulingv1.PodGroupPending),
+			},
+			Pods: []*v1.Pod{
+				// pending pod with owner1, under ns:c1/q:c1
+				util.BuildPod("c1", "p1", "", v1.PodPending, api.BuildResourceList("3", "1G"), "pg1", make(map[string]string), make(map[string]string)),
+				// pending pod with owner2, under ns:c1/q:c2
+				util.BuildPod("c1", "p2", "", v1.PodPending, api.BuildResourceList("1", "1G"), "pg2", make(map[string]string), make(map[string]string)),
+			},
+			Queues: []*schedulingv1.Queue{
+				util.BuildQueue("c1", 1, nil),
+				util.BuildQueue("c2", 1, nil),
+			},
+		},
+	}
+
+	trueValue := true
+	tiers := []conf.Tier{
+		{
+			Plugins: []conf.PluginOption{
+				{
+					Name:            drf.PluginName,
+					EnabledJobOrder: &trueValue,
+				},
+				{
+					Name:              proportion.PluginName,
+					EnabledQueueOrder: &trueValue,
+				},
+				{
+					Name: sla.PluginName,
+					Arguments: map[string]interface{}{
+						"sla-waiting-time": "3m",
+					},
+					EnabledJobOrder: &trueValue,
+				},
+				{
+					Name:            gang.PluginName,
+					EnabledJobOrder: &trueValue,
+				},
+			},
+		},
+	}
+	for i, test := range tests {
+		t.Run(test.Name, func(t *testing.T) {
+			test.Plugins = plugins
+			test.RegisterSession(tiers, nil)
+			defer test.Close()
+			action := New()
+			test.Run([]framework.Action{action})
+			if err := test.CheckAll(i); err != nil {
+				t.Fatalf("case: %s, action: %s, error: %s", test.Name, action.Name(), err)
```

Comment:
err string has include case index and case name.

Reply:
done